tool/gocross/gocross-wrapper.sh

```diff
@@ -26,40 +26,29 @@ repo_root="${BASH_SOURCE%/*}/../.."
 # being invoked from somewhere else.
 cd "$repo_root"
 
-toolchain="$HOME/.cache/tailscale-go"
+# toolchain, set below is the root of the Go toolchain we'll use to build
+# gocross.
+#
+# It's set to either something with a leading slash (if that's what
+# go.toolchain.rev contains, for testing), or otherwise in the common case it'll
+# be "$HOME/.cache/tsgo/GITHASH" where GITHASH is the contents of the
+# go.toolchain.rev file and the git commit of the
+# https://github.com/tailscale/go release artifact to download.
+toolchain=""
 
-if [[ -d "$toolchain" ]]; then
-    # A toolchain exists, but is it recent enough to compile gocross? If not,
-    # wipe it out so that the next if block fetches a usable one.
-    want_go_minor=$(grep -E '^go ' "go.mod" | cut -f2 -d'.')
-    have_go_minor=""
-    if [[ -f "$toolchain/VERSION" ]]; then
-        have_go_minor=$(head -1 "$toolchain/VERSION" | cut -f2 -d'.')
-    fi
-    # Shortly before stable releases, we run release candidate
-    # toolchains, which have a non-numeric suffix on the version
-    # number. Remove the rc qualifier, we just care about the minor
-    # version.
-    have_go_minor="${have_go_minor%rc*}"
-    if [[ -z "$have_go_minor" || "$have_go_minor" -lt "$want_go_minor" ]]; then
+read -r REV <go.toolchain.rev
+case "$REV" in
+/*)
+    toolchain="$REV"
+    ;;
+*)
+    toolchain="$HOME/.cache/tsgo/$REV"
+    if [[ ! -f "$toolchain.extracted" ]]; then
```

Comment:
I feel like this can lead to two extractions running concurrently and cause problems but that was already an existing problem so I don't think it's any worse than before.

Reply:
yeah I didn't even try to guard against that. doing it robustly with kernel involvement and without dangling stale lock files around would require some research (for me at least) to do portably

Reply:
Probably something like a unique directory name and symlink you can atomically slide into place

Reply:
oh, you care about two concurrently trampling each other, not preventing two from concurrently running and doing duplicate curl download work?

Reply:
The way I'd do it in future is to look for the dir, print to stdout that maybe there's another running, and wait for the .extracted, maybe wiht a timeout of some time like a few minutes

Reply:
I'll be happy to type something like the above into a follow up PR

Reply:
> oh, you care about two concurrently trampling each other, not preventing two from concurrently running and doing duplicate curl download work?
> 

Ideally both but at least the second one is wasted work without breaking correctness